spec/lib/synced/synchronizer_spec.rb

```diff
@@ -2,44 +2,34 @@
 
 describe Synced::Synchronizer do
   let(:account) { Account.create }
+  let(:model_class) { Amenity }
 
-  describe "#perform" do
-    context "for model without only_updated" do
-      it "chooses :full strategy" do
-        synchronizer = Synced::Synchronizer.new(Class, {})
-        expect(synchronizer.strategy).to be_an_instance_of(Synced::Strategies::Full)
-      end
+  context "without remote option" do
+    it "uses given :full strategy" do
+      synchronizer = Synced::Synchronizer.new(model_class, strategy: :full)
+      expect(synchronizer.strategy).to be_an_instance_of(Synced::Strategies::Full)
     end
 
-    context "for model without synced_all_at_key missing in the db" do
-      it "chooses :full strategy" do
-        synchronizer = Synced::Synchronizer.new(Class, {})
-        expect(synchronizer.strategy).to be_an_instance_of(Synced::Strategies::Full)
-      end
+    it "uses given :updated_since strategy" do
+      synchronizer = Synced::Synchronizer.new(model_class, strategy: :updated_since)
+      expect(synchronizer.strategy).to be_an_instance_of(Synced::Strategies::UpdatedSince)
     end
 
-    context "for model with only_updated and synced_all_at_key present in the db" do
-      it "chooses :updated_since strategy" do
-        synchronizer = Synced::Synchronizer.new(Class, only_updated: true,
-          synced_all_at_key: :synced_all_at)
-        expect(synchronizer.strategy).to be_an_instance_of(Synced::Strategies::UpdatedSince)
-      end
+    it "uses given :check strategy" do
+      synchronizer = Synced::Synchronizer.new(model_class, strategy: :check)
+      expect(synchronizer.strategy).to be_an_instance_of(Synced::Strategies::Check)
     end
+  end
 
-    context "for model with only_updated, synced_all_at_key and remote objects given" do
-      it "chooses :full strategy" do
-        synchronizer = Synced::Synchronizer.new(Class, only_updated: true,
-          synced_all_at_key: :synced_all_at, remote: [{}, {}])
-        expect(synchronizer.strategy).to be_an_instance_of(Synced::Strategies::Full)
-      end
+  context "without remote option" do
```

Comment:
I think it should be `with remote option`, no? `remote` option not being nil forces `full` strategy if I got it correctly.